junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/TemplateTest.java

```diff
@@ -147,4 +150,27 @@ public static void checkTuples() {
                 observedTuples);
     }
 
+    private static Stream<Version> versions() {
+        return Stream.of(
+                version("latest")
+        // TODO: waiting for new versions support in ozangunalp repo https://github.com/ozangunalp/kafka-native/issues/21
+        // version("3.3.1"),
+        // version("3.2.1")
+        );
+    }
+
+    static Set<String> observedVersions = new HashSet<>();
+
+    @AfterAll
+    public static void checkVersions() {
+        assertEquals(Set.of("latest-snapshot"), observedVersions);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void testVersions(@DimensionMethodSource("versions") @KRaftCluster TestcontainersKafkaCluster cluster)
+            throws Exception {
+        observedVersions.add(cluster.getKafkaVersion());
```

Comment:
Why are we adding the observed version to a set and asserting the values in `afterAll` especially when there is only one version?

Even in a future when multiple versions are supported (I'd argue we should update this test when that becomes a reality rather than build for it now) the current implementation doesn't work. We would want to assert that every version generated by `versions` should be returned by `cluster.getKafkaVersion`.

Reply:
Sorry but I don't see why it does not work for multiple versions.

I prefer to have it everything ready for the next change rather than leave it and having to make a big modification later on. But this is only my opinion.

Reply:
It doesn't work as implemented anyway. The problem as I see it is as things stand you end up with duplicate definitions of versions one in the source method and one in the afterAll

Reply:
Making assertions in `afterAll` seems highly questionable. Let's not do that.

* using `afterAll` for assertions violates the Principle of Least Surprise.
* you are asserting the combined effect of the running of all the test, but how do you know the developer actually ran all the tests?  He could be passing invoking Test#runSingle or selecting tests from the IDE.  The failure will likely confuse him.

Reply:
I think @franvila is just re-using the pattern I established (above in the source code) for being able to assert that a `@TestTemplate`-annotated method really generated the expected the expected set of tests. I agree that making the assertion in `afterAll` is not normal. The reason I did it was because I couldn't find a better way of being able to make that assertion, using the facilities provided by junit jupiter, when testing a junit extension. 

Perhaps a better way, which might work, would be to move the test for `@TestTemplate` to the `integration-test` module, and do something like the `ExceptionalTest` for the templates.

Reply:
That's right, I just tried to follow the same pattern established in the class. We could refactor this class if needed, but I think we should create another issue to fix that, because it is out of the scope of this PR, if you all agree.